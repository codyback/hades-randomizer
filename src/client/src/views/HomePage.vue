<template>
  <div>
    <h1>{{ heading }}</h1>
  </div>
</template>

<script lang="ts" setup>
defineProps({
  heading: {
    type: String,
    default: 'Hades - Randomizer',
  },
});
</script>
