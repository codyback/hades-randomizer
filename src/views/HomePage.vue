<template>
  <div>
    <h1>Hades - Randomizer</h1>
  </div>
</template>

<script></script>
