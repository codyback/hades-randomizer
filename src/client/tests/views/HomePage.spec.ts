import { mount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';
import HomePage from '../../src/views/HomePage.vue';

const mountWith = () => mount(HomePage, {
  shallow: true,
});

describe('views:HomePage', () => {
  it('renders correctly', () => {
    const wrapper = mount(HomePage);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders correctly - with prop', () => {
    const wrapper = mountWith();
    wrapper.setProps({ heading: 'Hello World!' });
    expect(wrapper.element).toMatchSnapshot();
  });
});
