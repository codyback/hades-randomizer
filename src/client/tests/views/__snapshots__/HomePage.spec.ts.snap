// Vitest Snapshot v1

exports[`views:HomePage > renders correctly - with prop 1`] = `
<div>
  <h1>
    Hello World!
  </h1>
</div>
`;

exports[`views:HomePage > renders correctly 1`] = `
<div>
  <h1>
    Hades - Randomizer
  </h1>
</div>
`;
